docs/sorting-algorithms/circle-sort.mdx

```diff
@@ -0,0 +1,284 @@
+---
+title: Circle Sort
+slug: /sorting-algorithms/circle-sort
+---
+
+## Introduction
+
+Circle sort is an in-place, recursive sorting algorithm. It can be visualized by drawing concentric circles on an array and the elements lying on the same circle diametrically opposite to each other are compared. If the value of the left element is greater than the value of the right element, then the elements are swapped. This process is repeated in a recursive manner where the array is divided into sub-arrays forming pairs of sorted elements until the whole array is sorted.
+
+## Explanation
+
+Given an unsorted array `A[]`. In layman's terms what circle sort does is, compares the first element of the array with the last element, the second element with the second last element, and so on. The array is then divided into two parts, and the same thing is repeated in both parts. This recursion goes on until the parts can't be divided any further.
+
+Let us have one example: 
+```
+Let given array A = [4, 5, 3, 6, 2, 1]
+The sorted array is A' = [1, 2, 3, 4, 5, 6]
+
+Let us walk through the process of circle sort.
+
+Compare first element and last element of the array
+- [4, 5, 3, 6, 2, 1] -> [1, 5, 3, 6, 2, 4] (Swapping first and last element)
+
+Compare second element and second last element of the array
+- [1, 5, 3, 6, 2, 4] -> [1, 2, 3, 6, 5, 4] (Swapping the elements)
+
+Compare third element and third last element of the array
+- [1, 2, 3, 6, 5, 4] -> [1, 2, 3, 6, 5, 4]
+
+- Now, A is divided into two halves.
+
+A1 = [1, 2, 3], A2 = [6, 5, 4]
+
+- Same steps are repeated for both the halves.
+
+A1 = [1, 2, 3]
+
+Compare first element and last element of the array
+- [1, 2, 3] -> [1, 2, 3]
+
+A2 - [6, 5, 4]
+
+Compare first element and last element of the array
+- [6, 5, 4] -> [4, 5, 6] (Swapping the elements)
+
+- Again A1 and A2 are divided.
+
+A1 = [1], A2 = [2, 3], A3 = [4], A4 = [5, 6]
+
+A2 = [2, 3]
+
+Compare first element and last element of the array
+- [2, 3] -> [2, 3]
+
+A4 = [5, 6]
+
+Compare first element and last element of the array
+- [5, 6] ->[5, 6]
+
+- Again A2 and A4 are divided.
+
+A1 = [1], A2 = [2], A3 = [3], A4 = [5], A6 = [6]
+
+- Now, each sub-array contain a single element. The algorithm stops here. And the array A is
+  now sorted.
+```
+
+## Algorithm
+
+Given an unsorted array `A[]`.
+
+  - Compare first element of the array to the last element, second element to the second last element, and so on.
+  - Split the array into two halves and recursively repeat the same steps on both the halves.
+  - The array is sorted when the sub-arrays can not be split any further, which means that each sub-array contains only one element.
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'C', value: 'c', },
```

Comment:
Maybe having an implementation in Java/Python would be preferred

Reply:
The code for the same was not present in NeoAlgo. Can I add it later sometime?

Reply:
Yes. Can you add the picture in meanwhile and it will be ready to merge I guess.